Suggestion widget vinit

vinitbytheminute opened the pull request:
Trello Card explaining Widget: https://trello.com/c/qQTwMjBs
## Description
 Created a Suggestion/Widget component that will allow users to send either a
1. Question about the Tenting Application that is not addressed in the FAQ page
2. Suggestion on how to improve the application

## Motivation and Context
Allowing users to send general feedback and question is an important feature to have. Not only do we want to keep our users happy but also they may provide us with a suggestion that can be very useful. 

## How Has This Been Tested?
1. Log in or Create an Account 
2. Go to Home page and scroll down
3. Press on "Tenting Application Question and Feedback Form"
4. Press on Questions or Suggestions


## Screenshots (if appropriate):
<img width="1196" alt="screen shot 2019-01-02 at 12 54 41 pm" src="https://user-images.githubusercontent.com/44485851/50612085-a43ab800-0e8d-11e9-894d-876f2118444b.png">
<img width="702" alt="screen shot 2019-01-02 at 12 54 50 pm" src="https://user-images.githubusercontent.com/44485851/50612087-ab61c600-0e8d-11e9-8bbc-29af99972604.png">
<img width="712" alt="screen shot 2019-01-02 at 12 54 58 pm" src="https://user-images.githubusercontent.com/44485851/50612093-b157a700-0e8d-11e9-8771-5bac5331eb37.png">



## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

vinitbytheminute committed:
Create basic UI of Question/Suggestion Widget

Embeded google form questionare into 'message' Accordion dropdown. 
Issue, cant access,edit, submit embed googel form.

vinitbytheminute committed:
Not Working: Embed Question Google Form into Widget

Unable to display embedded Form in embedquestoinForm function in the the 
Question object within the level1Panels on line 7. Working through this 
bug right now.

vinitbytheminute committed:
Catching up to staging Commits

vinitbytheminute committed:
Fix Suggestion and Question Forms UI and funtionality for widget 

Able to address commit (e1ff887) issues. Both subAccordians google forms 
are displayed and functional. The previous issue was Accordion.Accordion 
prop panels only takes an array of props. Originally, I was passing a 
fucntion that consist of both forms to that array. Now, I pass the 
arrays directly into the array as a value to content attribute.

vinitbytheminute committed:
Test Local branch and Remote Branch in sync

I recently changed my local branch from "Suggestion-Vinit" to 
"question-suggestion-widget-vinit". Now I would like to update my remote 
branch as well and ensure the remote and local branch are properly 
working.

vinitbytheminute committed:
Publich local branch remotely 

Since the previous commit(f0f4cf8) did not work, now I am making a new 
local branch and publishing this remotely. Then I will delete both 
'Suggestion-Vinit' and question-suggestion-widget-vinit".

vinitbytheminute committed:
Update Title of Widget 

Changed "Title from Tenting Application FAQ and Feeback" to "Tenting 
Application Question and Feedback Form"

Abdulla-Shahid commented:
Hey Vinit, 
Great Job on the Widget. So for the current design of the homepage, pending the addition of the upcoming game API/widget, the suggestion widget could fit great here, underneath the upcoming games widget (if we want to keep it on the home page): 

![49062819_652739508632575_3692222628040278016_n](https://user-images.githubusercontent.com/22209648/50679085-db18e700-0fcf-11e9-8518-e41816665b6f.png)

Let me know what you think, if you like it, I'll add it to the grid after I'm done working with Aman on the points widget!

mafuvadze commented:
- I think it looks very odd to have an embedded Google forms page in the website. I think it would be cleaner if you just redirected users trying to submit feedback to a separate page. 
- Maybe don't put the feedback widget on the home screen? I think it fits better as a footer link to another page. Or maybe as a nav bar tab.

mafuvadze commented on app/javascript/src/components/home/HomeBody/FeedbackForm.jsx:
```diff
@@ -0,0 +1,30 @@
+const level1Content = (
+  <div>
+    Disclaimer: This section is for feedback on and questions you may have about the GTHC tenting application
+    <strong> NOT </strong> on the Tenting Process itself. Please direct those questions to linemonitors@gmail.com
```
Make this email address a hotlink so they can email the line monitors by clicking on it

Vinitowinit replied:
Great suggestion. Just linked it

mafuvadze commented on app/javascript/src/components/home/HomeBody/FeedbackForm.jsx:
```diff
@@ -0,0 +1,30 @@
+
+
+const level1Panels = [
+  { key: 'panel-1a', title: 'Suggestions', content:{content: <iframe src="https://docs.google.com/forms/d/e/1FAIpQLScPghAkeRi-Ey0JhWRc2xmuG5fn6TofbIf147gsgIjhsRnuCQ/viewform?embedded=true" width="500" height="300" frameborder="0" marginheight="0" marginwidth="0" align='middle' >Loading...</iframe> }},
```
- Use more descriptive keys instead of panel-1a/b
 - Minor nit, but one thing that helps with readability is splitting json entities into separate lines

Vinitowinit replied:
1. I adjusted so that the key values are more descriptive. Thank you for the suggestion
2. I split the JSON entities as you requested.

Vinitowinit commented:
@mafuvadze I do see your point of view of having a widget for this may seem strange. I am open to any suggestions. I like it in the grid portion in the home page @Abdulla-Shahid is creating. I am open to anything. I guess we can talk about this the next meeting in a day or so and finalize this up.

vinitbytheminute committed:
Address Anesu Review and Make Appropriate Changes

I addressed Anesus three suggestions: made headers and key values more 
descriptive, made Json values more readable, and hot linked hlm email. I 
also added the component to the grid component in homebody/index.js. The 
placement of this widget is still questionable.

vinitbytheminute committed:
Merge branch 'staging' of https://github.com/GTHC/web into suggestion-widget-vinit

vinitbytheminute committed:
Update Branch to Staging 

I git pull origin staging so that this branch consist of all the new 
additions to staging to avoid merge conflicts later on.

mafuvadze reviewed (approved):
Great work

amanmibra commented:
I can resolve the conflicts on free time

amanmibra committed:
Merge branch 'staging' of github.com:GTHC/web into suggestion-widget-vinit

amanmibra committed:
Changes to code structure and added buttons

amanmibra reviewed (changes requested):
Vinit, I made some changes myself that you should look at. First, I made some code structure changes. Fixed some syntax, spacing, etc. Then, I decided it would best that we did not use `iframe` components for any component running on our app. This is because iframes are notoriously known for hindering performance and they look really ugly on our page as well tbh, so, I have replaced them with buttons. It doesn't have to buttons, but buttons was the only good alternative I could think of.

Go ahead and take a look at my most recent commit (just scroll up and click the recent commit message to see the changes of that commit **ONLY**) to see the exact change I made.

Once you decide which alternative you will use to iframes, whether it is buttons or not (as long as it opens the form on a separate page), please remove all of the lines of code that you had related to iframes.

amanmibra commented:
The widget with buttons

![image](https://user-images.githubusercontent.com/13587407/50737897-dbfe7400-119b-11e9-9bf9-6591cc0fc2c3.png)

vinitbytheminute committed:
Review Aman Changes/Suggestions to PR

The Iframes have been removed. I also changed the color of the buttons 
to "primary"(blue) to match the color scheme of the page.

amanmibra commented on app/javascript/src/components/home/HomeBody/widgets/FeedbackForm.jsx:
```diff
@@ -7,23 +7,6 @@ const suggestionURL = "https://docs.google.com/forms/d/e/1FAIpQLScPghAkeRi-Ey0Jh
-];
-
 //Incorporates an Accordian within the larger Accordian description or content
 const level1Content = (
```
Do not forget to remove this

amanmibra reviewed (approved):
Looks good, but change one more thing! I added a comment.
